src/main/java/chess/domain/attribute/Square.java

```diff
@@ -0,0 +1,84 @@
+package chess.domain.attribute;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import chess.domain.chessboard.attribute.Direction;
+
+public class Square {
+
+    private static final Map<String, Square> SQUARES = new HashMap<>();
```

Comment:
static final이지만 가변 컬렉션이니 상수 컨벤션의 네이밍은 아니었으면 좋겠네요. 

```suggestion
    private static final Map<String, Square> squares = new HashMap<>();
```

Reply:
아 직전 리뷰는 잊어주세요. 싱글톤처럼 만들어서 쓰시려는 의도가 있군요.

lazy loading으로 처리해주셨는데, 멀티 스레딩 환경에서 loading이 동시에 일어나면 서로 간섭할 가능성이 있습니다.
객체 초기화 단계에서 미리 세팅해주시는 게 나을거 같아요.

```suggestion
    private static final Map<String, Square> SQUARES = new HashMap<>();
    
    static {
        initializeSquares();
    }
```

Reply:
네 수정했습니다 :)  혹시, loading이 동시에 일어나서 서로 간섭하는 경우는 어떤 경우일까요? 예시가 궁금합니다!

Reply:
이전 코드가 기억 안 나는데, lazy loading이라고 표현해놓은 리뷰 기반으로 생각해보면...

우연히 서로 다른 두 개의 스레드에 의해 객체 생성이 동시에 일어난 경우입니다 ㅎ 
static SQUARES는 공유 객체 이므로 여러 스레드가 동시에 접근할 수 있습니다.
이 때 SQUARES가 생성 되었는지 검증하는 코드는 동시에 접근한 두 스레드 모두 통과하고, 내부에서 initializeSquares(); 를 동시에 수행하게 될거에요. 그러면 64개가 아닌 128개의 객체가 될 수 도 있습니다 (물론 이 객체에선 Key가 String이어서 그럴 일은 없지만요)

저희는 웹 개발자이므로 멀티 스레드가 매우 일상적인 환경이고, 멀티스레딩 환경에서의 공유 객체의 위험성에 대해 인식하고 있어야 합니다.